actors/evm/src/interpreter/instructions/call.rs

```diff
@@ -233,3 +225,65 @@ pub fn call<'r, BS: Blockstore, RT: Runtime<BS>>(
     stack.push(U256::from(1));
     Ok(())
 }
+
+pub fn callactor<'r, BS: Blockstore, RT: Runtime<BS>>(
+    state: &mut ExecutionState,
+    platform: &'r System<'r, BS, RT>,
+) -> Result<(), StatusCode> {
+    let ExecutionState { stack, memory, .. } = state;
+    let rt = &*platform.rt; // as immutable reference
+
+    // stack: GAS DEST VALUE METHODNUM INPUT-OFFSET INPUT-SIZE
+    // NOTE: we don't need output-offset/output-size (which the CALL instructions have)
+    //       becase these are kinda useless; we can just use RETURNDATA anyway.
+    // NOTE: gas is currently ignored
+    let _gas = stack.pop();
+    let dst = stack.pop();
+    let value = stack.pop();
+    let method = stack.pop();
+    let input_offset = stack.pop();
+    let input_size = stack.pop();
+
+    let input_region = get_memory_region(memory, input_offset, input_size)
+        .map_err(|_| StatusCode::InvalidMemoryAccess)?;
+
+    let result = {
+        let dst_addr = Address::try_from(dst)?
+            .as_id_address()
+            .ok_or_else(|| StatusCode::BadAddress(format!("not an actor id address: {}", dst)))?;
+
+        if method.bits() > 64 {
+            return Err(StatusCode::ArgumentOutOfRange(format!("bad method number: {}", method)));
+        }
+        let methodnum = method.as_u64();
+
+        let input_data = if let Some(MemoryRegion { offset, size }) = input_region {
+            &memory[offset..][..size.get()]
+        } else {
+            &[]
+        }
```

Comment:
Just use `unwrap_or_default()` here:

```suggestion
        let input_data = input_region
            .map(|MemoryRegion { offset, size }| &memory[offset..][..size.get()])
            .unwrap_or_default();
```

Reply:
uhm, more scala.....

What's up with the war on `IF`? Programming is impossible without it.

Also note that the compiler will have to apply smarts to ultimately generate the code we are substituting, so the only thing achieved by this so called "idiomatic" nonsense is obfuscation and ever increasing compile times....

Reply:
It's not Scala, it's Rust. You have an Option here and a type that implements Default. So doing this is the natural way. The Rust compiler optimises for code like this, but we can check with https://godbolt.org if you're in doubt (not sure if it supports Wasm).

Reply:
that doesnt resolve either obfuscation or ever increasing compile times ;)

Reply:
so, I just to be clear here: I am not getting git blamed for this. If you feel too strongly about it, feel free to apply the obfuscator in the EXT* pr :)

Reply:
Oh, and don't forget the environment :laughing: 
Then I guess we are already using a small nation's energy output to compile our warez, so that's just a drop in the bucket.